extensions/ollama/src/ollama-extension.ts

```diff
@@ -0,0 +1,115 @@
+/**********************************************************************
+ * Copyright (C) 2025 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import { type Disposable, type ExtensionContext, type Provider, provider } from '@kortex-app/api';
+import { createOllama } from 'ollama-ai-provider-v2';
+
+export class OllamaExtension {
+  #extensionContext: ExtensionContext;
+  #currentModels: string[] = [];
+  #connectionDisposable: Disposable | undefined;
+  #interval: NodeJS.Timeout | undefined;
+
+  constructor(extensionContext: ExtensionContext) {
+    this.#extensionContext = extensionContext;
+  }
+
+  async activate(): Promise<void> {
+    const ollamaProvider = provider.createProvider({
+      name: 'Ollama',
+      status: 'unknown',
+      id: 'ollama',
+      images: {
+        icon: './icon.png',
+        logo: {
+          dark: './icon.png',
+          light: './icon.png',
+        },
+      },
+    });
+    this.#extensionContext.subscriptions.push(ollamaProvider);
+
+    await this.updateModelsAndStatus(ollamaProvider);
+    this.#interval = setInterval(() => {
+      this.updateModelsAndStatus(ollamaProvider).catch((error: unknown) => {
+        console.error('Error updating Ollama models and status:', error);
+      });
+    }, 30000);
+  }
+
+  protected async updateModelsAndStatus(ollamaProvider: Provider): Promise<void> {
+    let models: Array<{ name: string }> = [];
+    let running = true;
+    try {
+      const res = await fetch('http://localhost:11434/api/tags');
+      if (!res.ok) throw new Error('Ollama not running');
```

Comment:
suggestion to make error more generic, because it could also be an internal server error, even with Ollama running, and res.ok would be false